lineapy/api/api.py

```diff
@@ -99,20 +111,37 @@ def save(reference: object, name: str) -> LineaArtifact:
 
     # serialize value to db if we haven't before
     # (happens with multiple artifacts pointing to the same value)
+    serialize_method = ARTIFACT_STORAGE_BACKEND.lineapy
     if not db.node_value_in_db(
         node_id=value_node_id, execution_id=execution_id
     ):
 
         # TODO add version or timestamp to allow saving of multiple pickle files for the same node id
-        # pickles value of artifact and saves to filesystem
-        pickle_name = _pickle_name(value_node_id, execution_id)
-        _try_write_to_pickle(reference, pickle_name)
+
+        artifact_serialize_metadata = serialize_artifact(
+            value_node_id,
+            execution_id,
+            reference,
+            name,
+            storage_backend,
+            **kwargs,
+        )
+        if (
```

Comment:
Have some mild concerns here with how this function will look if we have a bunch of serializers. 

This block of if statements to get artifact_path, and the block of code at the end to write to `mlflow_artifact_storage` will probably exist for every backend, which can make this function really really long. 

Let's spend some time here thinking if we can make the return value for `serialize_artifact` something that makes this easier to check. I think maybe some type of dataclass will be cleaner?